src/services/test_results/TestResultsBuilder.ts

```diff
@@ -1,64 +1,154 @@
 import {TestResultErrorAttributes} from '../../dto/test_results/TestResultErrorAttributes';
 import {TestResultBuildAttributes} from '../../dto/test_results/TestResultBuildAttributes';
 import {TestResultTestFieldAttributes} from '../../dto/test_results/TestResultTestFieldAttributes';
 import {TestResult} from '../../dto/test_results/TestResult';
 import {TestResultTestField} from '../../dto/test_results/TestResultTestField';
-import {TestResultTestRunElement} from '../../dto/test_results/TestResultTestRunElement';
+import {UnitResultElement} from '../../dto/test_results/UnitResultElement';
 import {TestResultError} from '../../dto/test_results/TestResultError';
 import {WebApi} from 'azure-devops-node-api';
 import * as ba from 'azure-devops-node-api/BuildApi';
 import * as ta from 'azure-devops-node-api/TestApi';
 import {TestCaseResult} from "azure-devops-node-api/interfaces/TestInterfaces";
 import {LogUtils} from "../../LogUtils";
+import {GherkinResultElement} from "../../dto/test_results/GherkinResultElement";
+import {GherkinResultData} from "../../dto/test_results/GherkinResultData";
+import {GherkinScenarioData} from "../../dto/test_results/GherkinScenarioData";
 
 let convert = require('xml-js');
 let xmlescape = require('xml-escape');
+const globby = require('globby');
+const {readFileSync} = require('fs-extra');
 
 export class TestResultsBuilder {
+    private static readonly CUCUMBER_TEST_TYPE = "Cucumber";
+    private static readonly GHERKIN_OCTANE_TEST_LEVEL_DISPLAY_VALUE = "Gherkin Test";
+    private static readonly UNIT_OCTANE_TEST_LEVEL_DISPLAY_VALUE = "Unit Test";
 
-    public static buildTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
+    public static buildUnitTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
         if (!testResults || !testResults.length) {
-            logger.info('No test results were retrieved/found');
+            logger.info('No unit test results were retrieved/found');
             return null;
         }
-        let testResultTestRuns = this.buildTestResultTestRun(testResults, logger);
+
+        let testResultTestRuns = this.buildUnitResultElement(testResults, logger);
         let testResultTestFields = this.buildTestResultTestFields(testResults[0].automatedTestType);
-        let testResultBuild = this.buildTestResultBuild(server_id, job_id, testResults[0].build.id);
+        let testResultBuild = new TestResultBuildAttributes(server_id, job_id, testResults[0].build.id);
+
+        return new TestResult(testResultBuild, testResultTestFields, testResultTestRuns);
+    }
+
+    public static buildGherkinTestResult(testResults: GherkinResultData[], server_id: string, job_id: string, build_id: string, logger: LogUtils): TestResult {
+        if (!testResults || !testResults.length) {
+            logger.info('No gherkin test results were retrieved/found');
+            return null;
```

Comment:
There should not exist something like this. Normally, I would create an EmptyTestResult, and return it, and process it as empty collection. The API will be more homogenous in this case. How this null is processed by the getTestResultXml, because it seems that you pass it to TestResult that afterwards is converted to xml.